lib/dal/src/func/associations.rs

```diff
@@ -25,10 +31,97 @@ pub enum FuncAssociationsError {
 
 type FuncAssociationsResult<T> = Result<T, FuncAssociationsError>;
 
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributePrototypeArgumentView {
+    pub func_argument_id: FuncArgumentId,
+    pub id: AttributePrototypeArgumentId,
+    pub input_socket_id: Option<InputSocketId>,
+}
+
+impl AttributePrototypeArgumentView {
+    pub async fn assemble(
```

Comment:
Logic in progress... It's functional, passes tests and works in the frontend, but it likely needs work, especially since the data and how it's used is different in the new engine. We'll likely tackle this as we dive deeper into saving and executing funcs.